iroh/src/endpoint.rs

```diff
@@ -965,27 +965,30 @@ impl Endpoint {
     /// Returns an error if closing the magic socket failed.
     /// TODO: Document error cases.
     pub async fn close(&self, error_code: VarInt, reason: &[u8]) -> Result<()> {
```

Comment:
This is a separate topic, and ties in with @matheus23 's closing ideas.  But I think it is really un-ergonomic that we make this take any arguments.  We should provide some other method to close all connections and make this close connections with a default error code and reason if there are any left.